File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nonzeroindex =0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[nonzeroindex] = nums[i]
                nonzeroindex+=1
        for i in range(nonzeroindex, len(nums)):
            nums[i] = 0

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
       int nonZeroIndex = 0; // Initialize the index for placing non-zero elements
        
        // Iterate through the array
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                // If the current element is non-zero, move it to the next available position
                nums[nonZeroIndex] = nums[i];
                
                // Increment the index for placing non-zero elements
                nonZeroIndex++;
            }
        }
        
        // Fill the remaining positions with zeros
        for (int i = nonZeroIndex; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}